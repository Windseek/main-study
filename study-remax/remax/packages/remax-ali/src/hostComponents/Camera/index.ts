import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface CameraProps {
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  mode?: 'normal' | 'scanCode';
  devicePosition?: 'front' | 'back';
  flash?: 'auto' | 'on' | 'off';
  outputDimension?: '360P' | '540P' | '720P' | '1080P' | 'max';
  applyMicPermissionWhenInit?: boolean;
  frameSize?: 'small' | 'medium' | 'large';
  frameFormat?: 'rgba' | 'jpg';
  maxDuration?: number;
  onStop?: (e: any) => void;
  onError?: (e: any) => void;
  onScanCode?: (e: any) => void;
}

export const Camera = createHostComponent<CameraProps>('camera') as React.ComponentType<CameraProps>;
