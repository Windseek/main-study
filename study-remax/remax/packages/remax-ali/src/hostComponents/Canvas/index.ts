import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface CanvasProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  style?: React.CSSProperties;
  className?: string;
  width?: string | number;
  height?: string | number;
  disableScroll?: boolean;
  type?: string;
  onTap?: (e: any) => void;
  onTouchStart?: (e: any) => void;
  onTouchMove?: (e: any) => void;
  onTouchEnd?: (e: any) => void;
  onTouchCancel?: (e: any) => void;
  onLongTap?: (e: any) => void;
}
export const Canvas = createHostComponent<CanvasProps>('canvas') as React.ComponentType<CanvasProps>;
