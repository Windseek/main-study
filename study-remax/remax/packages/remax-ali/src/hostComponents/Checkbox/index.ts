import { createHostComponent } from '@remax/runtime';

export interface CheckboxProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  name?: string;
  className?: string;
  value?: string;
  checked?: boolean;
  disabled?: boolean;
  color?: string;
  onChange?: (e: any) => void;
}

export const Checkbox = createHostComponent<CheckboxProps>('checkbox') as React.ComponentType<CheckboxProps>;
