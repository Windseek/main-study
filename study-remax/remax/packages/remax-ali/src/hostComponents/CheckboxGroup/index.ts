import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface CheckboxGroupProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  name?: string;
  onChange?: (e: any) => void;
}

export const CheckboxGroup = createHostComponent<CheckboxGroupProps>(
  'checkbox-group'
) as React.ComponentType<CheckboxGroupProps>;
