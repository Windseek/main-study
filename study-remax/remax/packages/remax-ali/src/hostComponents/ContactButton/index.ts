import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface ContactButtonProps {
  readonly dataset?: DOMStringMap;
  tntInstId: string;
  scene: string;
  size?: string | number;
  color?: string;
  icon?: string;
  alipayCardNo?: string;
}

export const ContactButton = createHostComponent<ContactButtonProps>(
  'contact-button'
) as React.ComponentType<ContactButtonProps>;
