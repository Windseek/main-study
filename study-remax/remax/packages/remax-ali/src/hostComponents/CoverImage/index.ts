import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface CoverImageProps {
  readonly dataset?: DOMStringMap;
  className?: string;
  style?: React.CSSProperties;
  src?: string;
  onTap?: (e: any) => void;
}

export const CoverImage = createHostComponent<CoverImageProps>('cover-image') as React.ComponentType<CoverImageProps>;
