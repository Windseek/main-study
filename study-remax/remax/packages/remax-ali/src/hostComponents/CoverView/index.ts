import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface CoverViewProps {
  readonly dataset?: DOMStringMap;
  className?: string;
  style?: React.CSSProperties;
  onTap?: (e: any) => void;
}

export const CoverView = createHostComponent<CoverViewProps>('cover-view') as React.ComponentType<CoverViewProps>;
