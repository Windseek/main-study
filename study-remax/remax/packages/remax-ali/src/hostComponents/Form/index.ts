import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface FormProps {
  readonly dataset?: DOMStringMap;
  className?: string;
  style?: React.CSSProperties;
  reportSubmit?: boolean;
  onSubmit?: (e: any) => void;
  onReset?: (e: any) => void;
}

export const Form = createHostComponent<FormProps>('form') as React.ComponentType<FormProps>;
