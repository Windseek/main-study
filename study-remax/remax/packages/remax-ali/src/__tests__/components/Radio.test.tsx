import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { Radio } from '../../hostComponents';

describe('Radio', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<Radio />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
