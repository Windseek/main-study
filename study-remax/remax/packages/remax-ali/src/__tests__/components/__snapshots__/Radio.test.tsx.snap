// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio render correctly 1`] = `<radio />`;
